src/api/app/controllers/webui/package_controller.rb

```diff
@@ -12,12 +12,12 @@ class Webui::PackageController < Webui::WebuiController
                              Regexp.new('\.arch$')].freeze
 
   before_action :set_project, only: [:show, :edit, :update, :index, :users, :dependency, :binary, :binaries, :requests, :statistics, :revisions,
-                                     :new, :branch_diff_info, :rdiff, :create, :save, :remove,
+                                     :new, :branch_diff_info, :rdiff, :create, :save, :remove, :live_build_log, :update_build_log,
                                      :remove_file, :save_person, :save_group, :remove_role, :view_file, :abort_build, :trigger_rebuild,
                                      :trigger_services, :wipe_binaries, :buildresult, :rpmlint_result, :rpmlint_log, :meta, :save_meta, :files]
 
   before_action :require_package, only: [:edit, :update, :show, :dependency, :binary, :binaries, :requests, :statistics, :revisions,
-                                         :branch_diff_info, :rdiff, :save, :save_meta, :remove,
+                                         :branch_diff_info, :rdiff, :save, :save_meta, :remove, :live_build_log, :update_build_log,
```

Comment:
requiring a package for build log would bring back the cash for remote packages again, no?

Reply:
`require_package` fetches the package with the same options for `Package.get_by_project_and_name` as what you pushed before. Previously, you switched the `use_source` option from `false` to `true`, but this option has a default value of `true`, see: https://github.com/openSUSE/open-build-service/blob/eeaaddc878c6c575030c6fff53596762465c621b/src/api/app/models/package.rb#L155-L157

The other options (`follow_project_links: true, follow_multibuild: true`) are the same in `require_package`. As for rescuing from `Package::ReadAccessError` and `Project::ReadAccessError`, those are already caught in `require_package` since both errors have the ancestor `APIError`:

https://github.com/openSUSE/open-build-service/blob/eeaaddc878c6c575030c6fff53596762465c621b/src/api/app/controllers/webui/webui_controller.rb#L139

```bash
# inside Rails console
Package::ReadAccessError.ancestors.include?(APIError)
=> true
Project::ReadAccessError.ancestors.include?(APIError)
=> true
```

I will test this further, but I think it's going to work.

Reply:
well, yes, default value is true, but setting it to false is wrong (you may of course also remove the extra parameter)